server/sonar-server/src/test/java/org/sonar/server/qualitygate/ws/GetByProjectActionTest.java

```diff
@@ -0,0 +1,201 @@
+
+    WsQualityGates.QualityGate qualityGate = result.getQualityGate();
+    assertThat(Long.valueOf(qualityGate.getId())).isEqualTo(dbQualityGate.getId());
+    assertThat(qualityGate.getName()).isEqualTo(dbQualityGate.getName());
+    assertThat(qualityGate.getDefault()).isTrue();
+  }
+
+  @Test
+  public void project_quality_gate_over_default() {
+    ComponentDto project = componentDb.insertComponent(newProjectDto());
+    QualityGateDto defaultDbQualityGate = insertQualityGate("Sonar way");
+    QualityGateDto dbQualityGate = insertQualityGate("My team QG");
+    setDefaultQualityGate(defaultDbQualityGate.getId());
+    associateProjectToQualityGate(project.getId(), dbQualityGate.getId());
+
+    GetByProjectWsResponse result = callByUuid(project.uuid());
+
+    WsQualityGates.QualityGate qualityGate = result.getQualityGate();
+    assertThat(qualityGate.getName()).isEqualTo(dbQualityGate.getName());
+    assertThat(qualityGate.getDefault()).isFalse();
+  }
+
+  @Test
+  public void get_by_project_key() {
+    ComponentDto project = componentDb.insertComponent(newProjectDto());
+    QualityGateDto dbQualityGate = insertQualityGate("My team QG");
+    associateProjectToQualityGate(project.getId(), dbQualityGate.getId());
+
+    GetByProjectWsResponse result = callByKey(project.key());
+
+    assertThat(result.getQualityGate().getName()).isEqualTo(dbQualityGate.getName());
+  }
+
+  @Test
+  public void fail_when_project_does_not_exist() {
+    expectedException.expect(NotFoundException.class);
+
+    callByUuid("Unknown");
+  }
+
+  @Test
+  public void fail_when_no_parameter() {
+    expectedException.expect(IllegalArgumentException.class);
+
+    call(null, null);
+  }
+
+  @Test
+  public void fail_when_project_uuid_and_key_provided() {
+    expectedException.expect(IllegalArgumentException.class);
+
+    call("uuid", "key");
+  }
+
+  private GetByProjectWsResponse callByUuid(String projectUuid) {
+    return call(projectUuid, null);
+  }
+
+  private GetByProjectWsResponse callByKey(String projectKey) {
+    return call(null, projectKey);
+  }
+
+  private GetByProjectWsResponse call(@Nullable String projectUuid, @Nullable String projectKey) {
+    TestRequest request = ws.newRequest()
+      .setMediaType(MediaTypes.PROTOBUF);
+
+    if (projectUuid != null) {
+      request.setParam(PARAM_PROJECT_ID, projectUuid);
+    }
+
+    if (projectKey != null) {
+      request.setParam(PARAM_PROJECT_KEY, projectKey);
+    }
+
+    InputStream response = request.execute().getInputStream();
+
+    try {
+      return GetByProjectWsResponse.parseFrom(response);
+    } catch (IOException e) {
+      throw propagate(e);
+    }
+  }
+
+  private QualityGateDto insertQualityGate(String name) {
+    QualityGateDto qualityGate = dbClient.qualityGateDao().insert(dbSession, new QualityGateDto().setName(name));
+    db.commit();
+    return qualityGate;
+  }
+
+  private void associateProjectToQualityGate(long componentId, long qualityGateId) {
```

Comment:
it's theorically possible but we want to have as few dependencies as possible. We set the data and QualityGates is a business oriented component. Possible in this specific case (as it's a dependency) but I prefer not to